src/main/java/com/client/core/base/tools/entitychanger/impl/StandardEntityChanger.java

```diff
@@ -0,0 +1,148 @@
+package com.client.core.base.tools.entitychanger.impl;
+
+import com.bullhornsdk.data.model.entity.core.type.BullhornEntity;
+import com.bullhornsdk.data.model.entity.embedded.Address;
+import com.client.core.base.tools.entitychanger.EntityChanger;
+import com.client.core.base.util.Utility;
+import org.apache.commons.lang3.ObjectUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+import org.joda.time.DateTime;
+import org.springframework.stereotype.Service;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.Map;
+
+@Service
+public class StandardEntityChanger implements EntityChanger {
+
+    private final Logger log = Logger.getLogger(getClass());
+
+    public StandardEntityChanger() {
+    }
+
+    @Override
+    public <T, V> V retrieveField(T entity, String field, Class<V> type) {
+        Object value = retrieveField(entity, field);
+
+        return asType(value, type);
+    }
+
+    @Override
+    public <T> Object retrieveField(T entity, String field) {
+        return getValue(entity, field);
+    }
+
+    @Override
+    public <T> T setField(T entity, String field, Object value) {
+        String[] fields = field.split("\\.");
+        String finalField = fields[fields.length - 1];
+        Object target = Arrays.stream(fields, 0, fields.length - 1).reduce(entity, (entityOrField, nextField) -> {
+            try {
+                Method getter = entityOrField.getClass().getDeclaredMethod(getterMethodName(nextField));
+                Class<?> fieldType = getter.getReturnType();
+                Method setter = entityOrField.getClass().getDeclaredMethod(setterMethodName(nextField), fieldType);
+                Object maybeValue = getter.invoke(entityOrField);
+                if (maybeValue == null) {
+                    setter.invoke(entityOrField, fieldType.newInstance());
+                }
+                return (T) getter.invoke(entityOrField);
+            } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException |
+                     InstantiationException e) {
+                throw new RuntimeException(e);
```

Comment:
Add a message with more than just the exception
```suggestion
                throw new RuntimeException("Error occurred during X phase", e);
```